hack/ensure-go.sh

```diff
@@ -38,7 +38,7 @@ EOF
   local go_version
   IFS=" " read -ra go_version <<< "$(go version)"
   local minimum_go_version
-  minimum_go_version=go1.20.4
+  minimum_go_version=go1.20
```

Comment:
Let's only enforce the right minor version here.

There is no kubekins image with Go 1.20.6 yet (only 1.20.5). But I don't want this to block our Go bump.

I think that's okay as the Go version of the kubekins image is only used for unit tests and linting (e.g. not for the image builds)